core_v14/src/metadata/mod.rs

```diff
@@ -61,6 +56,13 @@ pub struct Metadata {
     types: PortableRegistry,
 }
 
+/// Types are internally stored away in a type registry. Rather than exposing
+/// The scale-info logic, we store and hand back these pointers to the type
+/// information. This can be resolved into [`crate::substrate_type::SubstrateType`]
+/// when you'd like the full type information to work with.
+#[derive(Debug, Clone, Copy)]
+pub struct TypeId(u32);
```

Comment:
Couldn't we use [the `std` type](https://doc.rust-lang.org/src/core/any.rs.html#441-443) here?